runzi/execute/openquake/oq_hazard_task.py

```diff
@@ -176,6 +176,97 @@ def _sterilize_task_arguments_gmcmlt(self, ta):
+        else:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                if file_id := task_arguments["site_params"].get("locations_file_id"):
+                    headers = {"x-api-key": API_KEY}
+                    file_api = ToshiFile(
+                        API_URL,
+                        None,
+                        None,
+                        with_schema_validation=True,
+                        headers=headers,
+                    )
+                    file_api.download_file(file_id, target_dir=temp_dir, target_name="sites.csv")
+                    locations_file = Path(temp_dir) / "sites.csv"
+                else:
+                    locations_file = Path(task_arguments["site_params"]["locations_file"])
+                locations = get_locations([locations_file])
+                with locations_file.open() as lf:
+                    reader = csv.reader(lf)
+                    header = next(reader)
+                    if "vs30" in header:
+                        ind = header.index("vs30")
+                        vs30s = []
+                        for row in reader:
+                            vs30s.append(int(row[ind]))
+
+        backarc_flags = map(int, within_polygon(locations, backarc_polygon()))
+        if any(self.model.hazard_config.get_uniform_site_params()):
+            self.model.hazard_config.set_sites(locations, backarc=backarc_flags)
+        else:
+            self.model.hazard_config.set_sites(locations, vs30=vs30s, backarc=backarc_flags)
+
+    def set_hazard_curve_parameters(self, task_arguments: Dict[str, Any]):
+        """Set hazard curve for hazard curve calculation
+
+        Args:
+            task_arguments: a dict of task arguments
+        """
+        imts = task_arguments["hazard_curve"]["imts"]
+        imtls = task_arguments["hazard_curve"]["imtls"]
+        self.model.hazard_config.set_iml(imts, imtls)
+
+    def set_disagg_matrix_parameters(self, task_arguments: Dict[str, Any]):
+        """Set disagg matrix coordinates and point on hazard curve to disaggregate
+
+        Args:
+            task_arguments: a dict of task arguments
+        """
+        self.model.hazard_config.set_iml_disagg(
+            imt=task_arguments["hazard_curve"]["imt"], level=task_arguments["disagg"]["target_level"]
+        )
+        self.model.hazard_config.set_parameter("general", "calculation_mode", "disaggregation")
+        self.model.hazard_config.set_parameter(
+            "disaggregation", "disagg_outputs", " ".join(task_arguments["disagg"]["disagg_outputs"])
+        )
+        if mag_bin_width := task_arguments["disagg"]["mag_bin_width"]:
+            self.model.hazard_config.set_parameter("disaggregation", "mag_bin_width", str(mag_bin_width))
+        if distance_bin_width := task_arguments["disagg"]["distance_bin_width"]:
+            self.model.hazard_config.set_parameter("disaggregation", "mag_bin_width", str(distance_bin_width))
```

Comment:
Based on the pattern, should this be
```suggestion
            self.model.hazard_config.set_parameter("disaggregation", "distance_bin_width", str(distance_bin_width))
```

Reply:
good catch. that's why it would be good to have more test coverage